### PR TITLE
Branch 4450 water level

### PR DESCRIPTION
**Pull request by svelmurugan92**
- Added the sketch for Catena 4450 with water pressure sensor
- Added 0x16 format

### CODE REVIEW

File: catena4450m101_sensor/application_commands.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
+//
```
Comment (terrillmoore) [inline catena4450m101_sensor/application_commands.cpp]: This needs an MCCI standard header.

File: catena4450m101_sensor/application_commands.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+// By putting in a #included CPP file, we avoid the deep scan, and thereby
+// avoid the problem.
+//
+#include <Catena_CommandStream.h>
+#include <cstring>
+
+using namespace McciCatena;
+
+/* library routine -- but not in library yet! */
```
Comment (terrillmoore) [inline catena4450m101_sensor/application_commands.cpp]: Should we add this to the library now?

File: catena4450m101_sensor/application_commands.cpp (hunk 1 above)
Comment (terrillmoore) [inline catena4450m101_sensor/application_commands.cpp]: This really has nothing to do with water level. The changes for the added command need to be in a different pull request.

File: catena4450m101_sensor/catena4450m101_sensor.ino [hunk 3]
```diff
@@ -119,6 +120,9 @@ ThisCatena gCatena;
 //
 ThisCatena::LoRaWAN gLoRaWAN;
 
+// the transmit period
```
Comment (terrillmoore) [inline catena4450m101_sensor/catena4450m101_sensor.ino]: Once again, this belongs with previous file in different pull request.

File: catena4450m102_waterlevel/catena4450m102_waterlevel.ino [hunk 4]
```diff
@@ -0,0 +1,697 @@
...
+                        fFoundWaterTemp = false;
+                        }
+                else
+                        fFoundWaterTemp = true;
+                }
+
+        /* now, we kick off things by sending our first message */
+        gLed.Set(LedPattern::Joining);
+
+        // unit testing for the scaling functions
```
Comment (terrillmoore) [inline catena4450m102_waterlevel/catena4450m102_waterlevel.ino]: Delete these comments (line 361 though 377)

File: catena4450m102_waterlevel/catena4450m102_waterlevel.ino [hunk 5]
```diff
@@ -0,0 +1,697 @@
...
+				}   
+			}
+		
+		SenAD = (uint32_t) ((pBuffer[1] << 16) + (pBuffer[2] << 8) + pBuffer[3]);
+		gCatena.SafePrintf("SenAD: %u\n", SenAD);
+		
+		if (SenAD >= REFERENCE_VALUE) /* Check for negative value, If yes then print 0*/
+			{
+			SenPDat = (float)(SenAD - REFERENCE_VALUE) / CROSS_VALUE * RANGE;
+			Serial.print("SenPDat: ");
```
Comment (terrillmoore) [inline catena4450m102_waterlevel/catena4450m102_waterlevel.ino]: This will not work if there's no USB attached. Use safePrintf or similar.

File: extra/catena-message-0x16-format.md [hunk 6]
```diff
@@ -0,0 +1,528 @@
+# Understanding Catena 4450 format 0x16
```
Comment (terrillmoore) [inline extra/catena-message-0x16-format.md]: This doc also describes 0x14. The correct thing is to add this to the other md, rather than creating a new one.

Comment (terrillmoore) [reply extra/catena-message-0x16-format.md]: You must also update all the .js files to understand 0x16.

File: catena4450m101_sensor/application_commands.cpp (hunk 1 above)
Comment (svelmurugan92) [reply catena4450m101_sensor/application_commands.cpp]: Sure, I will pull the master and add only water level sensing related changes. Will do adding commands changes as separate pull request.

File: catena4450m102_waterlevel/catena4450m102_waterlevel.ino (hunk 4 above)
Comment (svelmurugan92) [reply catena4450m102_waterlevel/catena4450m102_waterlevel.ino]: Sure, Thank you

File: catena4450m102_waterlevel/catena4450m102_waterlevel.ino (hunk 5 above)
Comment (svelmurugan92) [reply catena4450m102_waterlevel/catena4450m102_waterlevel.ino]: Ok sure, Will use safePrintf

File: extra/catena-message-0x16-format.md (hunk 6 above)
Comment (svelmurugan92) [reply extra/catena-message-0x16-format.md]: Ok, I will merge the 0x14 format md file and 0x16 format md file. Also try to update js files.
